### ISSUE TITLE
Depedency conflicts with @react-native-firebase/crashlytics

### ISSUE THREAD

**Issue by thacio** [a7medev/react-native-ml-kit#61]
### What happened?

Failed building for ios when @react-native-firebase/crashlytics is present due to depency resolve.
Since I'm using EAS to build, I can't test changing the GoogleDataTransport (< 10.0, >= 9.4.1) requirement to try fixing that.

EAS error:
[!] CocoaPods could not find compatible versions for pod "FirebaseCoreExtension":
  In Podfile:
    RNFBAuth (from `../node_modules/@react-native-firebase/auth`) was resolved to 21.6.1, which depends on
      Firebase/Auth (= 11.5.0) was resolved to 11.5.0, which depends on
        FirebaseAuth (~> 11.5.0) was resolved to 11.5.0, which depends on
          FirebaseCoreExtension (= 11.5)
    RNFBCrashlytics (from `../node_modules/@react-native-firebase/crashlytics`) was resolved to 21.6.1, which depends on
      FirebaseCoreExtension
CocoaPods could not find compatible versions for pod "GoogleDataTransport":
  In Podfile:
    RNFBCrashlytics (from `../node_modules/@react-native-firebase/crashlytics`) was resolved to 21.6.1, which depends on
      Firebase/Crashlytics (= 11.5.0) was resolved to 11.5.0, which depends on
        FirebaseCrashlytics (~> 11.5.0) was resolved to 11.5.0, which depends on
          GoogleDataTransport (~> 10.0)
    RNMLKitTextRecognition (from `../node_modules/@react-native-ml-kit/text-recognition`) was resolved to 1.5.2, which depends on
      GoogleMLKit/TextRecognitionDevanagari (= 6.0.0) was resolved to 6.0.0, which depends on
        GoogleMLKit/MLKitCore (= 6.0.0) was resolved to 6.0.0, which depends on
          MLKitCommon (~> 11.0.0) was resolved to 11.0.0, which depends on
            GoogleDataTransport (< 10.0, >= 9.4.1)

### Version

@react-native-ml-kit/barcode-scanning: *version*@react-native-ml-kit/face-detection: *version*@react-native-ml-kit/identify-languages: *version*@react-native-ml-kit/image-labeling: *version*@react-native-ml-kit/text-recognition: *version*@react-native-ml-kit/translate-text: *version*

### Which ML Kit packages do you use?

- [ ] @react-native-ml-kit/barcode-scanning
- [ ] @react-native-ml-kit/face-detection
- [X] @react-native-ml-kit/identify-languages
- [ ] @react-native-ml-kit/image-labeling
- [X] @react-native-ml-kit/text-recognition
- [ ] @react-native-ml-kit/translate-text

### What platforms are you seeing this issue on?

- [ ] Android
- [X] iOS

### System Information

ystem:
  OS: macOS 14.5
  CPU: (8) arm64 Apple M2
  Memory: 344.08 MB / 8.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 20.18.1
    path: /usr/local/bin/node
  Yarn: Not Found
  npm:
    version: 10.8.2
    path: /usr/local/bin/npm
  Watchman: Not Found
Managers:
  CocoaPods:
    version: 1.16.2
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 24.1
      - iOS 18.1
      - macOS 15.1
      - tvOS 18.1
      - visionOS 2.1
      - watchOS 11.1
  Android SDK: Not Found
IDEs:
  Android Studio: Not Found
  Xcode:
    version: 16.1/16B40
    path: /usr/bin/xcodebuild
Languages:
  Java: Not Found
  Ruby:
    version: 3.3.6
    path: /opt/homebrew/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.74.5
    wanted: 0.74.5
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: Not found
  newArchEnabled: Not found
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found

### Steps to Reproduce

Install @react-native-firebase/crashlytics, @react-native-ml-kit/text-recognition and build with eas.

**Comment by adamalexander** [comment]
@thacio any fixes??

**Comment by thacio** [comment]
Not really. I uninstalled crashlytics, set all the firebase services to 21.3.0, and will use analytics meawhile. Considering later creating a class simulating crashlytics but using analytics instead, so it would be easier to migrate if this ever gets fixed.

**Comment by thacio** [comment]
> @thacio any fixes??
Well, good news and bad news.

The problem was GoogleMLKit. Other people were going through this problem, they solve it releasing the version 7.0.0.
https://github.com/googlesamples/mlkit/issues/877
So, I bumped every GoogleMLkit dependency to 7.0.0 as was able to compile with firebase 2.6.1 without a problem. Also, the deploymentTarget for ios increased to 15.5.0, so it needs to be added to expo app.json.

Now, the bad news is that it just stopped working altogether for me, but I can't say if it was the bump to 7.0.0 or something else. Since yesterday I've been trying to upgrade to expo sdk 52, so it might be another package that broke the functionalities. But probably it was something with react-native-mlkit, since if I don't upgrade it or firebase, it still works.

If you want to give it a shot, you can clone the repository, look for the GoogleMLKit and bump from 6.0.0 to 7.0.0. I've forged the repo with the bump. https://github.com/thacio/react-native-ml-kit
To install, you would need to clone it locally.
'''
git clone https://github.com/thacio/react-native-ml-kit
npm install ./react-native-ml-kit/text-recognition
'''

'''
{
  "expo": {
    "plugins": [
      [
        "expo-build-properties",
        {
          "ios": {
            "deploymentTarget": "15.5.0"
          }
        }
      ]
    ]
  }
}
'''

**Comment by adamalexander** [comment]
> > @thacio any fixes??
> > Well, good news and bad news.
> 
> The problem was GoogleMLKit. Other people were going through this problem, they solve it releasing the version 7.0.0. [googlesamples/mlkit#877](https://github.com/googlesamples/mlkit/issues/877) So, I bumped every GoogleMLkit dependency to 7.0.0 as was able to compile with firebase 2.6.1 without a problem. Also, the deploymentTarget for ios increased to 15.5.0, so it needs to be added to expo app.json.
> 
> Now, the bad news is that it just stopped working altogether for me, but I can't say if it was the bump to 7.0.0 or something else. Since yesterday I've been trying to upgrade to expo sdk 52, so it might be another package that broke the functionalities. But probably it was something with react-native-mlkit, since if I don't upgrade it or firebase, it still works.
> 
> If you want to give it a shot, you can clone the repository, look for the GoogleMLKit and bump from 6.0.0 to 7.0.0. I've forged the repo with the bump. https://github.com/thacio/react-native-ml-kit To install, you would need to clone it locally.
> 
> ```
> git clone https://github.com/thacio/react-native-ml-kit
> npm install ./react-native-ml-kit/text-recognition
> ```
> 
> ```
> {
>   "expo": {
>     "plugins": [
>       [
>         "expo-build-properties",
>         {
>           "ios": {
>             "deploymentTarget": "15.5.0"
>           }
>         }
>       ]
>     ]
>   }
> }
> ```
> ```

This is good news. I'll check it out.

**Comment by thacio** [comment]
Just confirming it, text_recognition breaks with GoogleMLKit 7.0.0

**Comment by prudhvi-techmo** [comment]
any solution?

**Comment by prudhvi-techmo** [comment]
any update on this issue? still facing issue

**Comment by Etaliya** [comment]
Any update on this issue, still facing

**Comment by Energieman** [comment]
The solution worked fine.
https://github.com/a7medev/react-native-ml-kit/issues/69#issuecomment-3199968324

**Comment by a7medev** [comment]
Hello @thacio, we just updated all packages to use the latest available ML Kit version. Can you please upgrade to the latest and see if that fixes your issue?

I'm closing this for now but feel free to reopen it if you're still facing the same issue.
